gridstatus/isone_api/isone_api.py

```diff
@@ -0,0 +1,439 @@
+                        f"Failed to get data from {url} with params: {api_params}"
+                    )
+                else:
+                    error_message = (
+                        f"Error: Failed to get data from {url} with params:"
+                        f" {api_params}"
+                    )
+                log.error(error_message)
+                response.raise_for_status()
+
+        if parse_json:
+            return response.json()
+        else:
+            return response.content
+
+    def get_locations(self) -> pd.DataFrame:
+        """
+        Get a list of all locations.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing location information.
+        """
+        url = f"{BASE_URL}/locations"
+        response = self.make_api_call(url)
+        if "Locations" not in response or "Location" not in response["Locations"]:
+            raise NoDataFoundException("No location data found.")
+
+        locations = response["Locations"]["Location"]
+        df = pd.DataFrame(locations)
+
+        return df
+
+    def get_location_by_id(self, location_id: int) -> pd.DataFrame:
+        """
+        Get information for a specific location by its ID.
+
+        Args:
+            location_id (int): The ID of the location to retrieve.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing the location information.
+        """
+        url = f"{BASE_URL}/locations/{location_id}"
+        response = self.make_api_call(url)
+
+        if "Location" not in response:
+            raise NoDataFoundException(f"No data found for location ID: {location_id}")
+
+        location = response["Location"]
+        df = pd.DataFrame([location])
+        return df
+
+    def get_locations_all(self) -> pd.DataFrame:
+        """
+        Get detailed information for all locations.
+
+        Returns:
+            pandas.DataFrame: A DataFrame containing detailed information for all locations.
+        """
+        url = f"{BASE_URL}/locations/all"
+        response = self.make_api_call(url)
+
+        if "Locations" not in response or "Location" not in response["Locations"]:
+            raise NoDataFoundException("No location data found.")
+
+        locations = response["Locations"]["Location"]
+        df = pd.DataFrame(locations)
+        return df
+
+    def _handle_demand(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Process demand DataFrame: convert types and rename columns.
+
+        Args:
+            df (pd.DataFrame): Input DataFrame with demand data.
+
+        Returns:
+            pd.DataFrame: Processed DataFrame.
+        """
+        df["BeginDate"] = pd.to_datetime(df["BeginDate"])
+        df["Load"] = pd.to_numeric(df["Load"], errors="coerce")
+        df["LocId"] = pd.to_numeric(df["LocId"], errors="coerce")
+        df = df.rename(columns={"BeginDate": "Interval Start"})
+        return df[["Interval Start", "Location", "LocId", "Load"]]
```

Comment:
we also want to add "Interval End". we do that across all datasets.

we also should rename `LocId` to `Location Id`. we like to avoid abbreviations and by adding space, the column name works better with case-insentive formatting